app/src/main/java/org/mozilla/fenix/settings/about/AboutPageStore.kt

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.about
+
+import mozilla.components.lib.state.Action
+import mozilla.components.lib.state.State
+import mozilla.components.lib.state.Store
+
+/**
+ * [AboutFragment]'s [Store]
+ * * Encompasses it's own:
+ *  - [State] for the View displayed in this Fragment.
+ *  - [Action]s mapping a user / system interaction to an intention to modify the above State.
+ *  - [Reducer]s for modifying the above State based on the above Actions.
+ *
+ *  @param initialState [AboutFragment] that will be shown initially to the user.
+ */
+class AboutPageStore(
+    initialState: AboutPageState
+) : Store<AboutPageState, AboutPageAction>(
+    initialState,
+    ::aboutPageReducer
+)
+
+/**
+ * [State] containing all data displayed to the user by this Fragment.
+ *
+ * @param items the list containing the header and all the items from About page.
+ */
+data class AboutPageState(
+    val items: List<AboutPageItem>
+) : State
+
+sealed class AboutPageItem {
+    data class Header(val aboutText: String, val content: String, val buildDate: String) : AboutPageItem()
+    data class Item(val type: AboutItem, val title: String) : AboutPageItem()
+}
+
+/**
+ * [Action] for [AboutFragment] changes. Currently there are no possible actions that can change
+ * the view's state.
+ */
+sealed class AboutPageAction : Action
+
+@Suppress("UNUSED_PARAMETER") // the action parameter is unused
+fun aboutPageReducer(
+    currentState: AboutPageState,
+    action: AboutPageAction
+): AboutPageState {
+    // There is no possible action that can change this View's state while it is displayed to the user.
```

Comment:
Hey @ValentinTimisica, I appreciate the work that has been put into this! But I think I'm leaning towards @NotWoods comment as well. It feels like we're adding a lot of complexity pulling in `lib-state` here for what is ultimately a static list that will probably never change. I like the forward thinking but after reading this comment this solution does start to feel a little heavy. We could probably accomplish all of this in just the AboutFragment/xml. I'm happy to talk about it more if you still feel strongly about it 😊